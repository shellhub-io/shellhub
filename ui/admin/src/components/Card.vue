<template>
  <v-hover v-slot="{ isHovering, props }">
    <v-card class="pa-2 mb-2" v-bind="props" :elevation="isHovering ? 24 : 8">
      <v-list-item :data-test="nameUseTest" lines="three">
        <div>
          <p class="text-overline mb-4" :title-test="title">
            {{ title }}
          </p>
          <v-list-item-title class="text-h4 mb-1" :stats-test="stats">
            {{ stats }}
          </v-list-item-title>
          <v-list-item-subtitle :content-test="content">
            {{ content }}
          </v-list-item-subtitle>
        </div>

        <v-spacer />

        <template v-slot:append>
          <div class="mt-4">
            <v-icon :icon="icon" size="x-large" :iconName-test="icon" />
          </div>
        </template>
      </v-list-item>

      <v-card-actions>
        <v-btn :to="pathName" color="dark" variant="text" tabindex="0" :buttonName-test="buttonName">
          {{ buttonName }}
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-hover>
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  props: {
    id: {
      type: Number,
      default: 0,
    },
    title: {
      type: String,
      default: "",
    },
    fieldObject: {
      type: String,
      default: "",
    },
    content: {
      type: String,
      default: "",
    },
    icon: {
      type: String,
      default: "",
    },
    buttonName: {
      type: String,
      default: "",
    },
    pathName: {
      type: String,
      default: "",
    },
    nameUseTest: {
      type: String,
      default: "",
    },
    stats: {
      type: Number,
      default: 0,
    },
  },
});
</script>
