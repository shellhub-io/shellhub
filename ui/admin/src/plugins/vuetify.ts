// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";
import { VNumberInput } from "vuetify/labs/VNumberInput";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
  components: {
    VNumberInput,
  },
});
