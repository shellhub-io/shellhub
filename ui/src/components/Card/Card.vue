<template>
  <v-hover v-slot="{ isHovering, props }">
    <v-card class="pa-1 mb-2 border bg-v-theme-surface" v-bind="props" :elevation="isHovering ? 24 : 8" density="compact" variant="flat">
      <v-list-item :data-test="nameUseTest" lines="three">
        <div>
          <p class="text-overline mb-1" :title-test="title">
            {{ title }}
          </p>
          <v-list-item-title class="text-h5 mb-1" :stats-test="stats">
            {{ stats }}
          </v-list-item-title>
          <v-list-item-subtitle :content-test="content">
            {{ content }}
          </v-list-item-subtitle>
        </div>

        <v-spacer />

        <template v-slot:append>
          <div class="mt-4">
            <v-icon :icon="icon" size="x-large" :iconName-test="icon" />
          </div>
        </template>
      </v-list-item>

      <v-card-actions>
        <v-btn
          v-if="buttonName != 'Add Device'"
          :to="pathName"
          color="primary"
          variant="elevated"
          tabindex="0"
          :buttonName-test="buttonName"
          class="text-subtitle"
          size="small"
        >
          {{ buttonName }}
        </v-btn>

        <DeviceAdd v-else size="small" />
      </v-card-actions>
    </v-card>
  </v-hover>
</template>

<script setup lang="ts">
import DeviceAdd from "../Devices/DeviceAdd.vue";

defineProps({
  id: {
    type: Number,
    default: 0,
  },
  title: {
    type: String,
    default: "",
  },
  fieldObject: {
    type: String,
    default: "",
  },
  content: {
    type: String,
    default: "",
  },
  icon: {
    type: String,
    default: "",
  },
  buttonName: {
    type: String,
    default: "",
  },
  pathName: {
    type: String,
    default: "",
  },
  nameUseTest: {
    type: String,
    default: "",
  },
  stats: {
    type: Number,
    default: 0,
  },
});
</script>

<style lang="scss">
  .border {
    border: thin solid rgba(255, 255, 255, 0.12);
  }
</style>
