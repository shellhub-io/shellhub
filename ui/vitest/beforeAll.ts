import MockAdapter from "axios-mock-adapter";
import { beforeAll } from "vitest";
import ResizeObserver from "resize-observer-polyfill";
import { systemApi } from "../src/api/http";

const mockedSystemInfo = {
  authentication: {
    local: false,
    saml: true,
  },
  endpoints: {
    api: "localhost:80",
    ssh: "localhost:22",
  },
  setup: true,
  version: "v0.18.3",
};

const systemApiMock = new MockAdapter(systemApi.getAxios());
systemApiMock.onGet("http://localhost:3000/info").reply(200, mockedSystemInfo);

global.CSS = { supports: () => false } as never;
global.ResizeObserver = ResizeObserver;

beforeAll(() => {
  global.CSS = {
    supports: () => false,
    escape: (str: string) => str,
  } as never;
});
