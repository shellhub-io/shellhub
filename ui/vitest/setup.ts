/**
 * Vitest setup function
 */
export async function setup() {
  global.CSS = {
    supports: () => false,
    escape: (str: string) => str,
  } as never;
}

// FAIL LOUDLY on unhandled promise rejections / errors
process.on("unhandledRejection", (reason) => {
  // eslint-disable-next-line no-console
  console.log("FAILED TO HANDLE PROMISE REJECTION");
  throw reason;
});

/**
 * Vitest Teardown function
 */
export async function teardown() {
  console.log("üìù vitest globalTeardown");
}
