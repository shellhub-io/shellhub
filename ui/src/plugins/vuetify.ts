// Styles
import "@mdi/font/css/materialdesignicons.css";
import "@fortawesome/fontawesome-free/css/all.css";
import "vuetify/styles";
import { VFileUpload, VFileUploadItem } from "vuetify/labs/VFileUpload";
import "../styles/_variables.scss";
import { aliases as faAliases, fa } from "vuetify/iconsets/fa";

// Vuetify
import { createVuetify } from "vuetify";

const light = {
  dark: false,
  colors: {
    primary: "#667acc",
    secondary: "#FFFFFF",
    background: "#F5F5F5",
    tabs: "#F5F5F5",
    foreground: "#F5F5F5",
    paymentForm: "#F5F5F5",
    terminal: "#0f1526",
    "v-theme-background": "#FFFFFF",
    "v-theme-surface": "#FFFFFF",
    "v-theme-card": "#FFFFFF",
  },
};

const dark = {
  dark: true,
  colors: {
    primary: "#667acc",
    secondary: "#1E2127",
    background: "#18191B",
    tabs: "#1E1E1E",
    foreground: "#1E1E1E",
    paymentForm: "#E0E0E0",
    terminal: "#0f1526",
    "v-theme-background": "#1E2127",
    "v-theme-surface": "#1E2127",
    "v-theme-card": "#22252B",
  },
};

export default createVuetify({
  components: {
    VFileUpload,
    VFileUploadItem,
  },
  theme: {
    defaultTheme: "dark",
    themes: {
      dark,
      light,
    },
  },
  defaults: {
    global: {

    },
    VTextField: {
      variant: "outlined",
      density: "comfortable",
    },
    VSelect: {
      variant: "outlined",
      density: "comfortable",
    },
    VTextarea: {
      variant: "outlined",
    },
  },
  icons: {
    faAliases,
    sets: {
      fa,
    },
  },
});
