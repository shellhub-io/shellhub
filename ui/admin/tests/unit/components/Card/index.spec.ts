import { createVuetify } from "vuetify";
import { mount } from "@vue/test-utils";
import { beforeEach, describe, expect, it } from "vitest";
import Card from "../../../../src/components/Card.vue";

describe("Card", () => {
  let wrapper;

  beforeEach(() => {
    const vuetify = createVuetify();

    wrapper = mount(Card, {
      props: {
        id: 0,
        title: "Registered Users",
        fieldObject: "registered_users",
        content: "Registered users",
        icon: "mdi-account-group",
        stats: 10,
        buttonName: "View all Users",
        pathName: "users",
        nameUseTest: "viewUsers-btn",
      },
      global: {
        plugins: [vuetify],
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper).toBeTruthy();
  });
  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Renders the template with data", () => {
    expect(wrapper.find('[data-test="viewUsers-btn"]').exists()).toBe(true);
  });

  it("Renders the title with data", () => {
    const title = wrapper.find('[title-test="Registered Users"]');
    expect(title.exists()).toBe(true);
    expect(title.text()).toBe("Registered Users");
  });

  it("Renders the stats with data", () => {
    const stats = wrapper.find('[stats-test="10"]');
    expect(stats.exists()).toBe(true);
    expect(stats.text()).toBe("10");
  });

  it("Renders the content with data", () => {
    const content = wrapper.find('[content-test="Registered users"]');
    expect(content.exists()).toBe(true);
    expect(content.text()).toBe("Registered users");
  });

  it("Renders the icon name with data", () => {
    const icon = wrapper.find('[iconName-test="mdi-account-group"]');
    expect(icon.exists()).toBe(true);
  });

  it("Renders the button name with data", () => {
    const buttonName = wrapper.find("[buttonName-test]");
    expect(buttonName.exists()).toBe(true);
    expect(buttonName.text()).toBe("View all Users");
  });
});
