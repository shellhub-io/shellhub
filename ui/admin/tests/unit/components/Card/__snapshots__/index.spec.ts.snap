// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > Renders the component 1`] = `
"<div class="v-card v-theme--light v-card--density-default elevation-8 v-card--variant-elevated pa-2 mb-2">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class="v-list-item v-theme--light v-list-item--density-default v-list-item--three-line rounded-0 v-list-item--variant-text" data-test="viewUsers-btn">
    <!----><span class="v-list-item__underlay"></span>
    <!---->
    <div class="v-list-item__content" data-no-activator="">
      <!---->
      <!---->
      <div>
        <p class="text-overline mb-4" title-test="Registered Users">Registered Users</p>
        <div class="v-list-item-title text-h4 mb-1" stats-test="10">10</div>
        <div class="v-list-item-subtitle" content-test="Registered users">Registered users</div>
      </div>
      <div class="v-spacer"></div>
    </div>
    <div class="v-list-item__append">
      <div class="mt-4"><i class="mdi-account-group mdi v-icon notranslate v-theme--light v-icon--size-x-large" aria-hidden="true" iconname-test="mdi-account-group"></i></div>
      <div class="v-list-item__spacer"></div>
    </div>
  </div>
  <div class="v-card-actions"><a class="v-btn v-btn--slim v-theme--light text-dark v-btn--density-default v-btn--size-default v-btn--variant-text" tabindex="0" buttonname-test="View all Users"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">View all Users</span>
      <!---->
      <!---->
    </a></div>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;
